### PR TITLE
GEODE-6621: ensure that getBucketOwners result is modifiable

### PR DESCRIPTION
**Pull request by dschneider-pivotal**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-core/src/main/java/org/apache/geode/internal/cache/ProxyBucketRegion.java [hunk 1]
```diff
@@ -352,10 +352,12 @@ public ProxyBucketRegion initialize() {
   @Override
   public Set<InternalDistributedMember> getBucketOwners() {
     Set<InternalDistributedMember> s = this.advisor.adviseInitialized();
+    if (s.isEmpty()) {
+      // getBucketOwners needs to return a modifiable set.
+      // adviseIntialized returns an unmodifiable set when it is empty.
```
Comment (jhuynh1) [inline geode-core/src/main/java/org/apache/geode/internal/cache/ProxyBucketRegion.java]: typo adviseIntialized is missing an i

Comment (dschneider-pivotal) [reply geode-core/src/main/java/org/apache/geode/internal/cache/ProxyBucketRegion.java]: fixed

**Comment by dschneider-pivotal** [comment]
I initially planned on adding a unit test to ProxyBucketRegionTest but it looked like ProxyBucketRegion would need to refactoring to get the advisor mockable. It uses a static method in the constructor to create the advisor and that static method is passed the partially constructed ProxyBucketRegion and uses some of its state to create the advisor. Since this is correcting a problem introduced by a refactoring that did not have tests, by undoing one of the changes, I think it is okay to not have test coverage for this pull request.

**Comment by dschneider-pivotal** [comment]
The one failure: "PersistentRecoveryOrderDUnitTest > testCrashDuringPreparePersistentId" is a known issue that Kirk is fixing.

**Pull request merged**